### PR TITLE
Feature/DynamoDBStreams: decompose process records

### PR DESCRIPTION
**Pull request by maxhoheiser**
## Motivation
Decompose function that processes dynamodb records so each individual record can be collected as an event by EventStudio before they are sent to the kinesis stream
Decompose function that submits process records to background process to be able to patch it in EventStudio and submit it with the parent thread context

### CODE REVIEW

**Review by giograno — CHANGES_REQUESTED** [review]
Changes look harmless 👍 I only have a few minor comments.

File: localstack-core/localstack/services/dynamodb/provider.py [hunk 1]
```diff
@@ -208,15 +208,26 @@ def forward_to_targets(
...
                 region_name=region_name,
                 records_map=records_map,
             )
         else:
             self._forward(account_id, region_name, records_map)
 
+    def _submit_records(
+        self, forwarder, account_id: str, region_name: str, records_map: RecordsMap
+    ):
+        "Required for patching submit with local thread context for EventStudio"
```
[the remark this answers is not in the document]
Comment (maxhoheiser) [reply localstack-core/localstack/services/dynamodb/provider.py]: fixed in: https://github.com/localstack/localstack/pull/11821/commits/66d704c2c7d8ccf07aee2ca30cc66c9e1e637b12

File: localstack-core/localstack/services/dynamodb/provider.py [hunk 2]
```diff
@@ -208,15 +208,26 @@ def forward_to_targets(
         self, account_id: str, region_name: str, records_map: RecordsMap, background: bool = True
     ) -> None:
         if background:
-            self.executor.submit(
-                self._forward,
+            self._submit_records(
+                forwarder=self._forward,
```
[the remark this answers is not in the document]
Comment (maxhoheiser) [reply localstack-core/localstack/services/dynamodb/provider.py]: fixed in: https://github.com/localstack/localstack/pull/11821/commits/5bc6ad8324fbaef4d433e37e5cf73b9e5d0a30ef

**Review by giograno — APPROVED** [review]
✔️

**Pull request merged**